LiteDB/Utils/Extensions/StringExtensions.cs

```diff
@@ -1,16 +1,25 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Runtime.CompilerServices;
 using System.Security.Cryptography;
 using System.Text;
+
 using static LiteDB.Constants;
 
 namespace LiteDB
 {
     internal static class StringExtensions
     {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static bool IsNullOrWhiteSpace(this string str)
         {
-            return str == null || str.Trim().Length == 0;
+            return string.IsNullOrWhiteSpace(str);
```

Comment:
Instead of calling the existing method, isn't better to replace all calls by this?

Reply:
The clr should inline it without aggressiveinlining, but i added it nontheless. https://web.archive.org/web/20120210222409/http://blogs.microsoft.co.il/blogs/sasha/archive/2012/01/20/aggressive-inlining-in-the-clr-4-5-jit.aspx

Reply:
![grafik](https://github.com/mbdavid/LiteDB/assets/11245306/bc7f5042-70c2-4f3f-aef6-acffecc9b992)
![grafik](https://github.com/mbdavid/LiteDB/assets/11245306/5add8573-e4dd-4cd3-94ad-e11f083ecdc8)

Here we see the content of IsNullOrWhiteSpace exists 2x: One in IsNullOrWhiteSpace and one in main
![grafik](https://github.com/mbdavid/LiteDB/assets/11245306/f586c724-13f3-4f01-9958-1a6d14e3b05b)
![grafik](https://github.com/mbdavid/LiteDB/assets/11245306/1e533e45-5497-4f85-9a88-70b6d6c46441)

Reply:
I'm aware about the AgressiveInling, but isn't guarantee that the compiler will inline it in all scenarios.
My question is more why keeping a method that will just forward to some method that exists on BCL, I'm fine to adding those but will be hard to point all devs to this extension methods

Reply:
I understand your point about the redundancy of having an extension method that essentially forwards to a BCL method. 
However, I believe there are a couple of reasons to keep our custom extension method:

- The extension method provides a level of convenience and readability for our codebase. It's a familiar and established pattern in the repository and helps maintain consistency.

- Changing all 50 or so calls to the method would introduce unnecessary churn and pollute the git history. It's better to keep the existing method and avoid making widespread changes for minimal benefit. I, atleast do not want to pollute this PR with that.

Additionally, I find the "".IsNullOrWhiteSpace() extension method more readable and it follows OOP principles, even if it's not strictly OOP. It makes the code more intuitive and aligns with the way we think about object-oriented design.

Anyway, i have no hard feelings if you want to change it, i would suggest we do it in a separate PR. I, for my part prefer a clean PR 😅

Reply:
As I said I'm fine to keeping this extension method for now. Even not seeing too much value on it.

> even if it's not strictly OOP

me neither (:

Reply:
IMO better replace all with the standard, because otherwise there will be both `string.IsNullOrWhiteSpace(str)` and `str.IsNullOrWhiteSpace()` calls in the codebase, which would be just a small inconsistency for new contributors, without a clear indication which one is preferred